spring-session/src/main/java/org/springframework/session/jdbc/JdbcOperationsSessionRepository.java

```diff
@@ -0,0 +1,483 @@
+					ps.setString(1, session.getId());
+					ps.setLong(2, session.getLastAccessedTime());
+					ps.setString(3, session.getPrincipalName());
+					JdbcOperationsSessionRepository.this.lobHandler.getLobCreator()
+							.setBlobAsBytes(ps, 4, sessionBytes);
+				}
+
+			});
+			session.setNew(false);
+			publishEvent(new SessionCreatedEvent(this, session));
+		}
+		else {
+			this.jdbcOperations.update(getQuery(Query.UPDATE_SESSION), new PreparedStatementSetter() {
+
+				@Override
+				public void setValues(PreparedStatement ps) throws SQLException {
+					ps.setLong(1, session.getLastAccessedTime());
+					ps.setString(2, session.getPrincipalName());
+					JdbcOperationsSessionRepository.this.lobHandler.getLobCreator()
+							.setBlobAsBytes(ps, 3, sessionBytes);
+					ps.setString(4, session.getId());
+				}
+
+			});
+		}
+	}
+
+	@Override
+	public JdbcSession getSession(String id) {
+		MapSession session = null;
+		try {
+			session = this.jdbcOperations.queryForObject(getQuery(Query.GET_SESSION),
+					new Object[] { id }, this.mapper);
+		}
+		catch (EmptyResultDataAccessException ignored) {
+		}
+
+		if (session != null) {
+			if (session.isExpired()) {
+				delete(id);
+			}
+			else {
+				return new JdbcSession(session);
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public void delete(String id) {
+		this.jdbcOperations.update(getQuery(Query.DELETE_SESSION), id);
+	}
+
+	@Override
+	public Map<String, JdbcSession> findByIndexNameAndIndexValue(String indexName, String indexValue) {
+		if (!PRINCIPAL_NAME_INDEX_NAME.equals(indexName)) {
+			return Collections.emptyMap();
+		}
+
+		List<MapSession> sessions = this.jdbcOperations.query(
+				getQuery(Query.LIST_SESSIONS_BY_PRINCIPAL_NAME), new Object[] { indexValue }, this.mapper);
+
+		Map<String, JdbcSession> sessionMap = new HashMap<String, JdbcSession>(sessions.size());
+
+		for (MapSession session : sessions) {
+			sessionMap.put(session.getId(), new JdbcSession(session));
+		}
+
+		return sessionMap;
+	}
+
+	@Scheduled(cron = "0 * * * * *")
+	public void cleanUpExpiredSessions() {
+		long now = System.currentTimeMillis();
+		long roundedNow = roundDownMinute(now);
+
+		if (logger.isDebugEnabled()) {
+			logger.debug("Cleaning up sessions expiring at " + new Date(roundedNow));
+		}
+
+		int deletedCount = this.jdbcOperations.update(
+				getQuery(Query.DELETE_SESSIONS_BY_LAST_ACCESS_TIME), roundedNow);
+
+		if (logger.isDebugEnabled()) {
+			logger.debug("Cleaned up " + deletedCount + " expired sessions");
+		}
+	}
+
+	/**
+	 * Replace patterns in the input to produce a valid SQL query. This implementation lazily initializes a
+	 * simple map-based cache, only replacing the table prefix on the first access to a named query. Further
+	 * accesses will be resolved from the cache.
+	 *
+	 * @param base the SQL query to be transformed
+	 * @return a transformed query with replacements
+	 */
+	protected String getQuery(Query base) {
```

Comment:
@rwinch we're very often in situation where customer owns the database and all we get is a schema with limited privileges, meaning we have multiple services that share the same schema. I agree it's a not great situation but its completely out our control.

Therefore I think allowing users to customize the table prefix is not some exotic requirement but rather a failry common thing we heavily rely on with other frameworks we use regulary:
- [Spring Integration](http://docs.spring.io/spring-integration/docs/4.2.5.RELEASE/reference/html/jdbc.html#jdbc-message-store-generic)
- [Spring Batch](http://docs.spring.io/spring-batch/trunk/reference/html/configureJob.html#repositoryTablePrefix)
- [Quartz Scheduler](http://quartz-scheduler.org/generated/2.2.1/html/qs-all/#page/Quartz_Scheduler_Documentation_Set%2Fre-jtx_jdbc_jobstoretx_configuration.html%23)

Not to drag this any further, I'll update the PR soon according to your preferences but expect another PR to provide the table prefix customization as soon as you merge this one :)